syntax = "proto3";

service Statistics {
    rpc GetStatisticsById (GetStatisticsByIdRequest) returns (GetStatisticsByIdReply) {}
    rpc GetTopPosts (GetTopPostsRequest) returns (GetTopPostsReply) {}
    rpc GetTopUsers (EmptyRequest) returns (GetTopUsersReply) {}
}

message GetStatisticsByIdRequest {
    int64 post_id = 1;
}

message GetStatisticsByIdReply {
    int64 post_id = 1;
    int64 likes = 2;
    int64 views = 3;
}

message GetTopPostsRequest {
    string flag = 1;
}

message PostReply {
    int64 post_id = 1;
    string login = 2;
    int64 amount = 3;
}

message GetTopPostsReply {
    repeated PostReply post_replies = 1;
}

message EmptyRequest {
}

message UserReply {
    string login = 1;
    int64 likes = 2;
}

message GetTopUsersReply {
    repeated UserReply user_replies = 1;
}
